test/specs/fontfaces.spec.mjs

```diff
@@ -0,0 +1,376 @@
+      });
+    });
+
+    describe("normal", () => {
+      it("should pick normal when exact match is available", () => {
+        const fontFaces = buildFontFaceMap([italic, oblique, normal]);
+
+        const result = resolveFontFace(fontFaces, [normal]);
+
+        expect(result).toEqual(normalizeFontFace(normal));
+      });
+
+      it("should prefer oblique over italic", () => {
+        const fontFaces = buildFontFaceMap([italic, oblique]);
+
+        const result = resolveFontFace(fontFaces, [normal]);
+
+        expect(result).toEqual(normalizeFontFace(oblique));
+      });
+
+      it("should use italic when neither normal or oblique is available", () => {
+        const fontFaces = buildFontFaceMap([italic]);
+
+        const result = resolveFontFace(fontFaces, [normal]);
+
+        expect(result).toEqual(normalizeFontFace(italic));
+      });
+    });
+  });
+
+  describe("font weight", () => {
+    describe("400 and 500", () => {
+      it("should match exact weight when 400", () => {
+        const fontFaces = buildFontFaceMap([w300, w400, w500]);
+
+        const result = resolveFontFace(fontFaces, [w400]);
+
+        expect(result).toEqual(normalizeFontFace(w400));
+      });
+
+      it("should match exact weight when 500", () => {
+        const fontFaces = buildFontFaceMap([w300, w400, w500]);
+
+        const result = resolveFontFace(fontFaces, [w400]);
+
+        expect(result).toEqual(normalizeFontFace(w400));
+      });
+
+      it("should try font-weight 500 first when desired weight is 400", () => {
+        const fontFaces = buildFontFaceMap([w300, w500]);
+
+        const result = resolveFontFace(fontFaces, [w400]);
+
+        expect(result).toEqual(normalizeFontFace(w500));
+      });
+
+      it("should pick smaller font-weight when desired weight is 400 and 500 is not available", () => {
+        const fontFaces = buildFontFaceMap([w100]);
+
+        const result = resolveFontFace(fontFaces, [w400]);
+
+        expect(result).toEqual(normalizeFontFace(w100));
+      });
+
+      it("should pick larger font-weight when no smaller is available", () => {
+        const fontFaces = buildFontFaceMap([w900]);
+
+        const result = resolveFontFace(fontFaces, [w400]);
+
+        expect(result).toEqual(normalizeFontFace(w900));
+      });
+
+      it("should try font-weight 400 first when desired weight is 500", () => {
+        const fontFaces = buildFontFaceMap([w600, w400]);
+
+        const result = resolveFontFace(fontFaces, [w500]);
+
+        expect(result).toEqual(normalizeFontFace(w400));
+      });
+
+      it("should pick larger font-weight when desired weight is 500 and 400 is not available", () => {
+        const fontFaces = buildFontFaceMap([w900]);
+
+        const result = resolveFontFace(fontFaces, [w500]);
+
+        expect(result).toEqual(normalizeFontFace(w900));
+      });
+
+      it("should pick larger font-weight when no smaller is available", () => {
```

Comment:
I think the description doesn't fit to the test case here.

Reply:
Yeah, it should be the reverse: `smaller font-weight when no larger is available`. There's another test-case with the same name on line 227, and this is basically the inverse of that.